### PR TITLE
Readability refactor

### PR DESCRIPTION
**Pull request by rgallardo-netflix**
Refactored code for readability. Split out most sections into self-contained methods with a single objective. Code is now longer, but less convoluted and with less anonymous nested things.

Cleaned up compiler warnings in main class and in the corresponding unit tests.

### CODE REVIEW

File: archaius2-core/src/main/java/com/netflix/archaius/ConfigProxyFactory.java [hunk 1]
```diff
@@ -182,131 +180,145 @@ interface MethodInvoker<T> {
...
-                    sbProperty.append("'");
-                    return sbProperty.toString();
-                }).collect(Collectors.joining(",")));
-                sb.append("]");
-                return sb.toString();
+            propertyNames.put(method, methodInvokerHolder.propertyName);
+
+            if (immutable) {
+                // Cache the current value of the property and always return that.
+                // Note that this will fail for parametrized properties!
```
Comment (akang31) [inline archaius2-core/src/main/java/com/netflix/archaius/ConfigProxyFactory.java]: There are a few instances where parameterized is misspelled parametrized

**Review by akang31 — APPROVED** [review]
Thanks for the changes! This makes ConfigProxyFactory much more self-explanatory.

File: archaius2-core/src/main/java/com/netflix/archaius/ConfigProxyFactory.java (hunk 1 above)
Comment (akang31) [reply archaius2-core/src/main/java/com/netflix/archaius/ConfigProxyFactory.java]: Oh it looks like both are valid, but I think it'd be best if we stuck with one or the other regardless

Comment (rgallardo-netflix) [reply archaius2-core/src/main/java/com/netflix/archaius/ConfigProxyFactory.java]: LOL, thanks for catching this!

**Pull request merged**